### PR TITLE
Fixes #37496 - RHEL registration template prioritizes paramaters

### PR DESCRIPTION
**Pull request by aburhinox**
Fixes #37496

This changes the priority on subman_org to use host_parm if present, then use katello generated org.  This will allow users to override the katello generated org through a host_parm, and if not present continue to work as today with the katello generated org.

### CODE REVIEW

**Review by stejskalleos — COMMENTED** [review]
IMO, it does make sense. We should allow users to override the output of the template by host parameters, but I'll leave the final ACK to @nofaralfasi

**Review by ekohl — APPROVED** [review]
@nofaralfasi is on PTO for this week, but it makes sense to me too. Shall we just merge it now?

**Review by nofaralfasi — APPROVED** [review]
LGTM.

**Comment by ekohl** [comment]
@aburhinox could you rephrase the the commit message to fit our convention? I also don't know why CI didn't run, so updating your commit may also solve that.

**Comment by aburhinox** [comment]
Thanks for taking a look at this.  I believe I followed the commit message guidance I found https://theforeman.org/handbook.html but it's still failing the automated tests.  Can anyone help point out what is wrong with the commit message?  I have the redmine number, added the short description, and have slightly longer description below, not sure why it's failing.

**Comment by stejskalleos** [comment]
4_commits
Hi,
It looks like you made a mistake while cleaning up the commit message. Right now, there are four commits instead of one.
How to fix it:

Make sure you have the latest `develop`:
* `git checkout develop`
* `git pull upstream develop`

Then rebase your branch and cleanup commits:
* `git checkout your-branch`
* `git rebase -i develop`, keep only this line:
```
reword d94fd9d Fixes #37496 

(remove all other commit lines)
```
* Save and exit editor
* Rename the commit message to "Fixes #ID - Your message"
* Save and exit editor

Check changes:
* `git diff develop`

After you check that everything is correct:
* `git push -f`

**Comment by aburhinox** [comment]
thanks @stejskalleos, that did the trick.  it looks like it's in good condition to review now.

**Review by stejskalleos — APPROVED** [review]

**Comment by stejskalleos** [comment]
Thanks @aburhinox
